src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,24 +93,71 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
-    kv = SecretClient(credential=csc, vault_url=kwargs['url'])
-    return kv.get_secret(
-        name=kwargs['secret_field'],
-        version=kwargs.get('secret_version', ''),
+def _initialize_credential(
+    tenant: str = '',
+    client: str = '',
+    secret: str = '',
+    /,
```

Comment:
I made a mistake in the initial suggested signature: https://wemake-python-styleguide.readthedocs.io/en/latest/pages/usage/violations/best_practices.html#wemake_python_styleguide.violations.best_practices.ProblematicFunctionParamsViolation. It doesn't really make sense to have a bunch of positional-only params with defaults.
```suggestion
```